### DISCUSSION TITLE
nspanel.yaml install error

### DISCUSSION THREAD

**Discussion by mcgnai** [Blackymas/NSPanel_HA_Blueprint#2262] [Q&A]
I have problems installing the nspanel.yaml file. Below I report the error during compilation. what could it depend on? Thank you

IINFO ESPHome 2024.8.1
INFO Reading configuration /config/esphome/nspanel.yaml...
INFO Detected timezone 'Europe/Rome'
INFO Generating C++ source...
INFO Compiling app...
Processing nspanel (board: esp32dev; framework: espidf; platform: platformio/espressif32@5.4.0)
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash

framework-espidf @ 3.40407.240606 (4.4.7)
tool-cmake @ 3.16.9
tool-ninja @ 1.10.2
toolchain-esp32ulp @ 2.35.0-20220830
toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch5
Reading CMake configuration...
-- The C compiler identification is GNU 8.4.0
-- The CXX compiler identification is GNU 8.4.0
-- The ASM compiler identification is GNU
-- Found assembler: /data/cache/platformio/packages/toolchain-xtensa-esp32/bin/xtensa-esp32-elf-gcc
-- Check for working C compiler: /data/cache/platformio/packages/toolchain-xtensa-esp32/bin/xtensa-esp32-elf-gcc
-- Check for working C compiler: /data/cache/platformio/packages/toolchain-xtensa-esp32/bin/xtensa-esp32-elf-gcc -- broken
-- Configuring incomplete, errors occurred!
See also "/data/build/nspanel/.pioenvs/nspanel/CMakeFiles/CMakeOutput.log".
See also "/data/build/nspanel/.pioenvs/nspanel/CMakeFiles/CMakeError.log".
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
CMake Error at /data/cache/platformio/packages/tool-cmake/share/cmake-3.16/Modules/CMakeTestCCompiler.cmake:60 (message):
The C compiler

"/data/cache/platformio/packages/toolchain-xtensa-esp32/bin/xtensa-esp32-elf-gcc"
is not able to compile a simple test program.

It fails with the following output:

Change Dir: /data/build/nspanel/.pioenvs/nspanel/CMakeFiles/CMakeTmp

Run Build Command(s):/data/cache/platformio/packages/tool-ninja/ninja cmTC_03776 && [1/2] Building C object CMakeFiles/cmTC_03776.dir/testCCompiler.c.obj
[2/2] Linking C executable cmTC_03776
FAILED: cmTC_03776 
: && /data/cache/platformio/packages/toolchain-xtensa-esp32/bin/xtensa-esp32-elf-gcc -mlongcalls -Wno-frame-address   CMakeFiles/cmTC_03776.dir/testCCompiler.c.obj  -o cmTC_03776   && :
/data/cache/platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld: /data/cache/platformio/packages/toolchain-xtensa-esp32/bin/../libexec/gcc/xtensa-esp32-elf/8.4.0/liblto_plugin.so: error loading plugin: /data/cache/platformio/packages/toolchain-xtensa-esp32/bin/../libexec/gcc/xtensa-esp32-elf/8.4.0/liblto_plugin.so: cannot open shared object file: No such file or directory
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
/data/cache/platformio/packages/framework-espidf/tools/cmake/project.cmake:381 (__project)
CMakeLists.txt:3 (project)

========================== [FAILED] Took 7.49 seconds ==========================

**Comment by edwardtfn** [comment]
Have you tried cleaning the build files?

https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/docs/error_compiling.md#clean-up-build-files

**Reply by mcgnai** [reply]
yes, I tried but it didn't solve the problem
